### ISSUE TITLE
BUG: image driver state file write fails intermittently 

### ISSUE THREAD

**Issue by jhamman** [UW-Hydro/VIC#536]
I'm fairly sure this bug was introduced in #494. This is contributing to the current failing travis tests.

I'm assigning @yixinmao since she knows the state file piece the best and I'll work with her to find a solution.

**Comment by bartnijssen** [comment]
Any information on how it fails? Error message? Seg fault?

**Comment by jhamman** [comment]
I think I'm passed the segfault behavior but I'm not exactly sure what the error is since the model is exiting in a few places right now.

The current failure on travis is raised in this line: https://github.com/UW-Hydro/VIC/blob/develop/vic/drivers/shared_image/src/vic_init_output.c#L296-L299

```
	--->---------------------------------------------------------------------------
	--->Traceback (most recent call last):
	--->/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xed) [0x7feac5df17ed]
	--->/home/travis/build/jhamman/VIC/vic/drivers/image/vic_image.exe() [0x4514d1]
	--->/home/travis/build/jhamman/VIC/vic/drivers/image/vic_image.exe() [0x48e067]
	--->/home/travis/build/jhamman/VIC/vic/drivers/image/vic_image.exe() [0x48e29c]
	--->[ERROR] ../shared_image/src/vic_init_output.c:299: errno: NetCDF: Invalid argument: Error creating /home/travis/workdir/VIC_tests_20160621/examples/Example-Image-Stehekin-base-case/results/fluxes.1949-01-01.nc 
```

However, I believe I was getting segfault behavior without the `NC_WRITE |` flag. Its going to take some sleuthing...

**Comment by jhamman** [comment]
closed via #537
